overlays/emacs.nix

```diff
@@ -73,17 +73,31 @@ let
         drv: drv.overrideAttrs (old:
           let
             libName = drv: super.lib.removeSuffix "-grammar" drv.pname;
-            lib = drv: ''lib${libName drv}.so'';
-            linkCmd = drv: "ln -s ${drv}/parser $out/lib/${lib drv}";
+            libSuffix = if super.stdenv.isDarwin then "dylib" else "so";
+            lib = drv: ''lib${libName drv}.${libSuffix}'';
+            linkCmd = drv:
+              if super.stdenv.isDarwin
+              then ''cp ${drv}/parser $out/lib/${lib drv}
+                     /usr/bin/install_name_tool -id $out/lib/${lib drv} $out/lib/${lib drv}
+                     /usr/bin/codesign -s - -f $out/lib/${lib drv}
+                ''
+              else ''ln -s ${drv}/parser $out/lib/${lib drv}'';
             linkerFlag = drv: "-l" + libName drv;
+
             plugins = args.withTreeSitterPlugins self.pkgs.tree-sitter-grammars;
             tree-sitter-grammars = super.runCommand "tree-sitter-grammars" {}
               (super.lib.concatStringsSep "\n" (["mkdir -p $out/lib"] ++ (map linkCmd plugins)));
           in {
             buildInputs = old.buildInputs ++ [ self.pkgs.tree-sitter tree-sitter-grammars ];
             # before building the `.el` files, we need to allow the `tree-sitter` libraries
             # bundled in emacs to be dynamically loaded.
-            TREE_SITTER_LIBS = super.lib.concatStringsSep " " ([ "-ltree-sitter" ] ++ (map linkerFlag plugins)); 
+            TREE_SITTER_LIBS = super.lib.concatStringsSep " " ([ "-ltree-sitter" ] ++ (map linkerFlag plugins));
+            # Add to directories that tree-sitter looks in for language definitions / shared object parsers
+            postPatch = old.postPatch + ''
+                 substituteInPlace src/treesit.c \
```

Comment:
I am guessing that after pushing paths in https://git.savannah.gnu.org/cgit/emacs.git/tree/src/treesit.c?h=64044f545add60e045ff16a9891b06f429ac935f#n533, the `dynlib_open` calls at https://git.savannah.gnu.org/cgit/emacs.git/tree/src/treesit.c?h=64044f545add60e045ff16a9891b06f429ac935f#n567 work as expected on Linux, while on macOS, the names are not found. Do you agree with my assessment, and if so, do you consider that kind of thing expected behavior on macOS?

Reply:
I agree with your assessment, assuming the linux build does indeed work, i've never tried , but i'm not sure what the expected behaviour of macOS is, not an expert here, it's possible there's a better way to do this that i'm not aware of, but I know this way works at least. I can add a darwin conditional if you'd like. I'll do a bit of googling on macos standard system load paths... but no guarantee.